server/src/controllers/users.js

```diff
@@ -7,9 +7,20 @@ require("dotenv").config();
 class UserController {
   // The currentUser route controller
   static currentUser = async (req, res) => {
+    // Check if the user has gmail authorized
+
+    const token = await db.Token.findOne({
+      where: {
+        userId: req.currentUser.id,
+      },
+    });
+    const userData = req.currentUser;
+
+    userData["hasGmailAuthorized"] = Boolean(token);
```

Comment:
In the future we can cache this on the user (denormalizing data and putting a column on the user for this field), instead of having to make two database calls everytime we fetch the user)